model/activist.go

```diff
@@ -348,6 +348,106 @@ func HideUser(db *sqlx.DB, userID int) error {
 	return nil
 }
 
+// Merge originalUserID into mergedUserID.
+//  - The original user is hidden
+//  - All of the original user's event attendance is updated to be the merged user.
+func MergeUser(db *sqlx.DB, originalUserID, mergedUserID int) error {
+	if originalUserID == 0 {
+		return errors.New("originalUserID cannot be 0")
+	}
+	if mergedUserID == 0 {
+		return errors.New("mergedUserID cannot be 0")
+	}
+	if originalUserID == mergedUserID {
+		return errors.New("originalUser and mergedUser cannot be the same")
+	}
+
+	tx, err := db.Beginx()
+	if err != nil {
+		return errors.Wrap(err, "could not create transaction")
+	}
+
+	_, err = tx.Exec(`UPDATE activists SET hidden = true WHERE id = ?`, originalUserID)
+	if err != nil {
+		tx.Rollback()
+		return errors.Wrapf(err, "failed to hide original activist %d", originalUserID)
+	}
+
+	// Let's do this the boring way.
+	var originalUserEventIDs []int
+	err = tx.Select(&originalUserEventIDs, `
+SELECT event_id
+FROM event_attendance
+WHERE
+  activist_id = ?
+`, originalUserID)
+	if err != nil {
+		tx.Rollback()
+		return errors.Wrapf(err, "failed to get original activist's events: %d",
+			originalUserID)
+	}
+	for _, eventID := range originalUserEventIDs {
+		// If the merged user also went to the event, just
+		// delete the original user's attendance record.
+		// Otherwise, update the original user's attendance
+		// record so that the activist id is the merged user.
+		var mergedUserAttended int
+		err = tx.Get(&mergedUserAttended, `
+SELECT count(*) FROM event_attendance WHERE activist_id = ? AND event_id = ?
```

Comment:
Related to my comment on the previous query excluding the "merged" User's duplicate events.. 

This wouldn't be needed, if we don't update or delete any of the "source" User's attendance records. By updating or deleting these records, it may complicate things when "undoing" a merge.

Perhaps we could handle this by just adding attendance records for the events (non-duplicate) found in the previous query. The "source" user is already being hidden, and we're tracking the merges. So whenever an undo merge feature is added, it would be just a couple simple operations.

1) Toggle the Activist's "hidden" field.
2) Remove the events that were merged into the "merged" Activist, based on the Activist ID that is getting "re-activated".

I usually try to avoid deletions when there's a possibility that they could be undone in the future.

Reply:
(One piece of context: hidden activists don't show up on the activist page, but they still get counted for attendance.)

I think our ideas are basically equivalent. You can completely reverse a merge with the info stored in merged_activist_attendance.

The one difference is that for the method you're suggesting, we'd need another field on activists (like `merged`) that indicates that the activist shouldn't show up on the activist page AND shouldn't be counted for attendance. And then when we count attendance, we'll need to join `event_attendance` and `activists` to only include unmerged activists.

Reply:
So a "hidden" activist still shows up in the list for taking attendance? What's the point of setting the hidden field when merging then?